### PR TITLE
`symbols-view` rolling fixes

### PR DESCRIPTION
**Pull request by savetheclocktower**
Better to keep two different PRs here than try to shoehorn these fixes into the Tree-sitter rolling PR. This one should also land in time for 1.114.

### CODE REVIEW

**Comment by claytonrcarter** [comment]
I was using symbols view today and kept wanting for `toggle-file-symbols` to copy the current selection (if any) into the finder @ toggle, in the same way that `toggle-file-finder` and `toggle-buffer-finder` do. Is it in-scope for this PR to make a feature request, or should I file an FR issue?

**Comment by savetheclocktower** [comment]
> I was using symbols view today and kept wanting for `toggle-file-symbols` to copy the current selection (if any) into the finder @ toggle, in the same way that `toggle-file-finder` and `toggle-buffer-finder` do. Is it in-scope for this PR to make a feature request, or should I file an FR issue?

Ah, I was unaware of that setting in `fuzzy-finder`. It's a fine idea, but the thing I ought to be doing is having **Symbols View: Go To Declaration** try to act on buffer symbols instead of project symbols when there are no providers of the latter. That way you could just put your cursor on something, jump to its definition in the same file, then jump back to where you were.

**Comment by claytonrcarter** [comment]
> that setting in `fuzzy-finder`

Oof. You know, I *did* go looking to see if there was a setting controlling this, but I totally overlooked that one. I'll move this out to an FR issue, since it doesn't feel relevant for a fix PR.

> act on buffer symbols instead of project symbols when there are no providers of the latter

Oh, that would explain why it wasn't working for me! 😄

**Comment by savetheclocktower** [comment]
OK, I don't see myself adding any new stuff before 1.114, so this one's ready for review.

**Comment by DeeDeeG** [comment]
I see you're interested in this landing for 1.114, but I feel at least as under-qualified to review this as I am for the tree-sitter stuff. If there's no one else to review it I can see what I can do between now and tomorrow evening in my timezone when Release day usually kicks off.

**Comment by savetheclocktower** [comment]
I'll ask around in Discord about it. You've done more than your fair share.

**Review by confused-Techie — APPROVED** [review]
This change looks great to me!
Overall it's full of fantastic stuff:
* Better iconography resolution (Although I hate that it has to be duplicated between `symbols-view` and `symbols-provider-tree-sitter` but understand that's the best way for now)
* Better documentation in the type file, which is extremely helpful if we need to point any more community package maintainers there
* Brand new feature of auto querying your selected text, which I love
* And obviously loads more safety mechanisms, lots more nullish assignment, which I'm always a fan of.

The only thing really here that struck me as odd is the manual assignment of `order` in the configuration schema, not that this is a bad thing, I've just personally never seen it done, honestly made me do a double take lol, but no problem there.

Otherwise, I'm not the most qualified to review the new query file changes, but I'll happily trust you on those, since well your wrote them all originally.

Plus we have tests that seem to provide full coverage on everything here, so I'll happily say lets ship it!

**Comment by savetheclocktower** [comment]
> The only thing really here that struck me as odd is the manual assignment of `order` in the configuration schema, not that this is a bad thing, I've just personally never seen it done, honestly made me do a double take lol, but no problem there.

Without it, your properties are sorted alphabetically. Less than ideal!

File: packages/symbols-view/lib/symbols-view.js [hunk 1]
```diff
@@ -23,6 +23,70 @@ function validateListControllerProps(props) {
   ));
 }
 
+// Assign a default icon type for each tag — or what LSP calls “kind.” This
+// list is copied directly from the LSP spec's exhaustive list of potential
+// symbol kinds:
+//
+// https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#symbolKind
+function iconForTag(tag) {
```
Comment (mauricioszabo) [inline packages/symbols-view/lib/symbols-view.js:31]: I have the same need on my `generic-lsp` package, maybe we can think about movint this to the `atom.ui` namespace API?

Comment (savetheclocktower) [reply packages/symbols-view/lib/symbols-view.js:31]: I wouldn't be opposed to that. Ideally, we figure out how to set a good default and allow the user to override it somehow.

**Pull request merged**